apisix/ssl.lua

```diff
@@ -153,6 +159,9 @@ local function aes_decrypt_pkey(origin, field)
     for _, aes_128_cbc_with_iv in ipairs(aes_128_cbc_with_iv_tbl) do
         local decrypted = aes_128_cbc_with_iv:decrypt(decoded_key)
         if decrypted then
+            if C.ERR_peek_error() then
+                C.ERR_clear_error()
+            end
```

Comment:
I feel that it is more appropriate to pop up an error when decryption fails. Currently, there may be two unexpected behaviors:
1. When all keys fail to decrypt, the resulting error will be stored in OpenSSL's error queue, which may affect subsequent error checks.
2. When the first key decrypts successfully, no actual error occurs during decryption, but we still pop all errors from the queue.

Reply:
In a single call, it may generate multiple openssl errors, and we should not be responsible for handling errors thrown by the library. Instead, we should ensure that no matter how many errors it generates, it should not affect normal requests.


There are primarily two scenarios:

1. For successful decryption (normal request): We do not need to care about the contents of the error queue; the error queue must be cleared.
2. For total decryption failure (failed request): In fact, nginx will clear the error queue before the handshake, so we don't need to handle it additionally.